walker.go

```diff
@@ -15,6 +15,7 @@ import (
 type WalkOpt struct {
 	IncludePaths    []string // todo: remove?
 	ExcludePatterns []string
+	Filter          func(*Stat) bool
```

Comment:
As it’s possible to mutate the Stat struct... I’m not sure that Filter is self descriptive enough. It’s more close to map/reduce in fact.

Reply:
Updated